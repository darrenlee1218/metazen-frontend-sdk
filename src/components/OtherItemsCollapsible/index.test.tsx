import React from 'react';
import { render, fireEvent, screen, cleanup } from '@testing-library/react';
import { ComponentProps } from 'react';
import { OtherItemsCollapsible } from '.';
import { TestProviders } from '@utils/testing/TestProviders';

afterEach(cleanup);

const defaultProps: ComponentProps<typeof OtherItemsCollapsible> = {
  label: 'Label',
  numItems: 10,
  isOpen: false,
  children: <h1>Child</h1>,
  onClick: () => {},
};

it('should not render anything if numItems is 0', () => {
  const props = { ...defaultProps, numItems: 0 };
  render(<OtherItemsCollapsible {...props} />, { wrapper: TestProviders });
  expect(screen.queryByText(defaultProps.label)).toBe(null);
});

it('should hide children when isOpen is false', () => {
  render(<OtherItemsCollapsible {...defaultProps} />, { wrapper: TestProviders });
  const el = screen.getByTestId('collapse');
  expect(el.className).toMatch(/MuiCollapse-hidden/i);
});

it('should not hide children if isOpen is true', () => {
  const props = { ...defaultProps, isOpen: true };
  render(<OtherItemsCollapsible {...props} />, { wrapper: TestProviders });
  const el = screen.getByTestId('collapse');
  expect(el.className).not.toMatch(/MuiCollapse-hidden/i);
});

it('should trigger callback after a click event is fired', () => {
  const callback = jest.fn();
  const props = { ...defaultProps, onClick: callback };
  render(<OtherItemsCollapsible {...props} />, { wrapper: TestProviders });

  fireEvent.click(screen.getByText(defaultProps.label));
  expect(callback).toHaveBeenCalled();
});

it('matches snapshot', () => {
  const { container } = render(<OtherItemsCollapsible {...defaultProps} />, { wrapper: TestProviders });
  expect(container).toMatchSnapshot();
});
