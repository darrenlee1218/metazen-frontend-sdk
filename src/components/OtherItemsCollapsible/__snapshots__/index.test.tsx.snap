// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium  css-m95ct2-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ie8d53-MuiTypography-root"
      >
        Label
      </h5>
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium css-1gnd1fd-MuiButton-endIcon"
      >
        <svg
          fill="none"
          height="12px"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="12px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 9L12 15L18 9"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </button>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-rr7dug-MuiDivider-root"
    />
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
      data-testid="collapse"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <h1>
            Child
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
